### ISSUE TITLE
A mistake caused by batch norm layers

### ISSUE THREAD

**Issue by PatrickHua** [PatrickHua/SimSiam#1]
Initially I used the `z1, z2 = encoder(torch.cat([x1, x2])).chunk(2)` to replace the twice forwarding in simsiam. However I realized the output can not be aligned with the original implementation in the paper. Here's a simplified example:

```python
import torch, torchvision
x1 = torch.randn((2,3,224,224))
x2 = torch.randn_like(x1)
encoder = torchvision.models.resnet50()

z1, z2 = encoder(x1), encoder(x2)
print(z1,z2)
z1, z2 = encoder(torch.cat([x1, x2])).chunk(2)
print(z1,z2)
```
This gives different outputs for z1 and z2.

Then I disabled the bn using eval():
```
encoder.eval()
z1, z2 = encoder(x1), encoder(x2)
print(z1,z2)
z1, z2 = encoder(torch.cat([x1, x2])).chunk(2)
print(z1,z2)
```
The outputs are the same now!
